Tests/UnitTests/MediaContextTests.swift

```diff
@@ -0,0 +1,197 @@
+        mediaContext.setChapterInfo(chapterInfo, metadata: chapterMetadata)
+        XCTAssertNotNil(mediaContext.chapterInfo)
+        XCTAssertEqual(4, mediaContext.chapterInfo?.toMap().count)
+
+        mediaContext.clearChapterInfo()
+        XCTAssertNil(mediaContext.chapterInfo)
+        XCTAssertTrue(mediaContext.chapterMetadata.isEmpty)
+    }
+
+    func testStartPlayerState_isInStateReturnsTrue() {
+        let mediaContext = MediaContext(mediaInfo: mediaInfo, metadata: mediaMetadata)
+        XCTAssertTrue(mediaContext.startState(info: muteStateInfo))
+        XCTAssertTrue(mediaContext.isInState(info: muteStateInfo))
+    }
+
+    func testStartPlayerState_failsAfterMaxNoOfStatesCreated() {
+        let mediaContext = MediaContext(mediaInfo: mediaInfo, metadata: mediaMetadata)
+        for stateName in 1...10 {
+            XCTAssertTrue(mediaContext.startState(info: StateInfo(stateName: "State\(stateName)")!))
+        }
+
+        XCTAssertFalse(mediaContext.startState(info: StateInfo(stateName: "State11")!))
+        XCTAssertFalse(mediaContext.startState(info: StateInfo(stateName: "State12")!))
+    }
+
+    func testClearStates_AfterMaxNoOfStatesCreated_clearAStates_allowsNewStates() {
+        let mediaContext = MediaContext(mediaInfo: mediaInfo, metadata: mediaMetadata)
+        for stateName in 1...10 {
+            XCTAssertTrue(mediaContext.startState(info: StateInfo(stateName: "State\(stateName)")!))
+        }
+
+        mediaContext.clearStates()
+
+        // Now can again track 10 new states
+        XCTAssertTrue(mediaContext.startState(info: StateInfo(stateName: "State11")!))
+        XCTAssertTrue(mediaContext.startState(info: StateInfo(stateName: "State12")!))
+    }
+
+    func testEndPlayerState_isInStateReturnsFalse() {
+        let mediaContext = MediaContext(mediaInfo: mediaInfo, metadata: mediaMetadata)
+        XCTAssertTrue(mediaContext.startState(info: muteStateInfo))
+        XCTAssertTrue(mediaContext.startState(info: testStateInfo))
+
+        mediaContext.endState(info: muteStateInfo)
+        XCTAssertFalse(mediaContext.isInState(info: muteStateInfo))
+        XCTAssertTrue(mediaContext.isInState(info: testStateInfo))
+
+        XCTAssertTrue(mediaContext.endState(info: testStateInfo))
+        XCTAssertFalse(mediaContext.isInState(info: testStateInfo))
+    }
+
+    func testGetAllStates_returnsAllStatesWithTheirCurrentStatus() {
+        let mediaContext = MediaContext(mediaInfo: mediaInfo, metadata: mediaMetadata)
+        for stateName in 1...10 {
+            XCTAssertTrue(mediaContext.startState(info: StateInfo(stateName: "State\(stateName)")!))
+        }
+
+        var activeStatesList = mediaContext.getActiveTrackedStates()
+        XCTAssertEqual(10, activeStatesList.count)
+        for stateName in 1...10 {
+            XCTAssertTrue(activeStatesList.contains(StateInfo(stateName: "State\(stateName)")!))  // 1-5 states are active
```

Comment:
this comment belongs to L171, here seems like all states should be active after they were started